bindings/localstorage/localstorage.go

```diff
@@ -0,0 +1,240 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package localstorage
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	securejoin "github.com/cyphar/filepath-securejoin"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/google/uuid"
+)
+
+const (
+	fileName = "fileName"
+)
+
+// LocalStorage allows saving files to disk
+type LocalStorage struct {
+	metadata *Metadata
+	logger   logger.Logger
+}
+
+// Metadata defines the metadata
+type Metadata struct {
+	RootPath string `json:"rootPath"`
+}
+
+// NewLocalStorage returns a new LocalStorage instance
+func NewLocalStorage(logger logger.Logger) *LocalStorage {
+	return &LocalStorage{logger: logger}
+}
+
+// Init performs metadata parsing
+func (ls *LocalStorage) Init(metadata bindings.Metadata) error {
+	m, err := ls.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	ls.metadata = m
+
+	err = os.MkdirAll(ls.metadata.RootPath, 0666)
```

Comment:
Check if it already exists first. Also, why 0666 and not simply 0600?

Reply:
After further investigation, I've set the perms to 0777 by default since the umask will be applied.